### PR TITLE
Pr/mpi tests

### PR DESCRIPTION
**Pull request by bd4**

### CODE REVIEW

File: xprof/xprof.rb.in [hunk 1]
```diff
@@ -236,6 +237,7 @@ class Sync_daemon
   end
 
   def finalize
```
Comment (TApplencourt) [inline xprof/xprof.rb.in:239]: I think you need to call it somewhere not?

File: .github/workflows/presubmit.yml [hunk 2]
```diff
@@ -255,3 +257,48 @@ jobs:
...
+      - run: make check
+        working-directory: build
+        env:
+          THAPI_VALGRIND: 1
+      - uses: actions/upload-artifact@v3
+        if: failure()
+        with:
+          name: build-and-check
+          path: |
+            build/**/*.log
```
[the remark this answers is not in the document]
Comment (bd4) [reply .github/workflows/presubmit.yml]: ya the check that sync_daemon is built can just be moved to the int test, saves a lot of time.

File: .github/workflows/presubmit.yml [hunk 3]
```diff
@@ -127,8 +127,40 @@ jobs:
         working-directory: build
       - run: make -j install
         working-directory: build
-      - run: |
-         IPROF=./build/ici/bin/iprof THAPI_TEST_BIN=clinfo bats integration_tests/test.bats
+      - run: ls -l bin/sync_daemon_mpi
+        working-directory: build/ici
```
Comment (TApplencourt) [inline .github/workflows/presubmit.yml:131]:
```suggestion
```

File: .github/workflows/presubmit.yml [hunk 4]
```diff
@@ -127,8 +127,40 @@ jobs:
...
+          bats integration_tests/sync_daemon_test.bats
+        env:
+          THAPI_BIN_DIR: ./build/ici/bin
+          THAPI_SYNC_DAEMON: fs
+      - name: test sync_daemon_mpi
+        run: |
+          bats integration_tests/sync_daemon_test.bats
+        env:
+          THAPI_BIN_DIR: ./build/ici/bin
+          THAPI_SYNC_DAEMON: mpi
```
Comment (TApplencourt) [inline .github/workflows/presubmit.yml:163]: Can we use that to simplify `https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs` ?

**Review by TApplencourt — APPROVED** [review]

**Pull request merged**